### DISCUSSION TITLE
Create3 topics do not respond to requests

### DISCUSSION THREAD

**Discussion by gjf2a** [iRobotEducation/create3_docs#397] [Questions] [answered]
### How are you connecting to your Create 3?

Wi-Fi (ROS 2)

### Computer(s) Model(s) and Operating System(s)

Libre Le Potato

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

H.1.2

### Which RMW is your robot running?

FastRTPS

### Does your robot have an assigned namespace? If so, please share.

/archangel

### Is the robot connected to a network? If so, what is the network type?

Yes, work network

### Are there multiple Create 3 robots connected to your network?

Yes

### Is multicast enabled?

Yes

### What is the Adapter Board's USB/BLE Toggle currently switched to?

USB

### Describe your question.

My robot normally gives a solid topic list on the second try after startup:

`ferrer@ferrer-aml-s905x-cc:~ros2 topic list
/archangel/battery_state
/archangel/cmd_audio
/archangel/cmd_lightring
/archangel/cmd_vel
/archangel/dock_status
/archangel/hazard_detection
/archangel/imu
/archangel/interface_buttons
/archangel/ir_intensity
/archangel/ir_opcode
/archangel/kidnap_status
/archangel/mobility_monitor/transition_event
/archangel/mouse
/archangel/odom
/archangel/robot_state/transition_event
/archangel/slip_status
/archangel/static_transform/transition_event
/archangel/stop_status
/archangel/tf
/archangel/tf_static
/archangel/wheel_status
/archangel/wheel_ticks
/archangel/wheel_vels
/parameter_events
/rosout
`

However, it does not consistently respond to publishing. For instance, when I give the following command:

`ros2 topic pub /cmd_vel geometry_msgs/Twist '{linear: {x: 1.0, y: 0.0, z: 0.0}, angular: {x: 0.0, y: 0.0, z: 0.0}}'`

It responds by giving a `publishing` message, but the robot never moves.

`ferrer@ferrer-aml-s905x-cc:~$ ros2 topic pub /cmd_vel geometry_msgs/Twist '{linear: {x: 1.0, y: 0.0, z: 0.0}, angular: {x: 0.0, y: 0.0, z: 0.0}}'
publisher: beginning loop
publishing #1: geometry_msgs.msg.Twist(linear=geometry_msgs.msg.Vector3(x=1.0, y=0.0, z=0.0), angular=geometry_msgs.msg.Vector3(x=0.0, y=0.0, z=0.0))

publishing #2: geometry_msgs.msg.Twist(linear=geometry_msgs.msg.Vector3(x=1.0, y=0.0, z=0.0), angular=geometry_msgs.msg.Vector3(x=0.0, y=0.0, z=0.0))

publishing #3: geometry_msgs.msg.Twist(linear=geometry_msgs.msg.Vector3(x=1.0, y=0.0, z=0.0), angular=geometry_msgs.msg.Vector3(x=0.0, y=0.0, z=0.0))

publishing #4: geometry_msgs.msg.Twist(linear=geometry_msgs.msg.Vector3(x=1.0, y=0.0, z=0.0), angular=geometry_msgs.msg.Vector3(x=0.0, y=0.0, z=0.0))

publishing #5: geometry_msgs.msg.Twist(linear=geometry_msgs.msg.Vector3(x=1.0, y=0.0, z=0.0), angular=geometry_msgs.msg.Vector3(x=0.0, y=0.0, z=0.0))

publishing #6: geometry_msgs.msg.Twist(linear=geometry_msgs.msg.Vector3(x=1.0, y=0.0, z=0.0), angular=geometry_msgs.msg.Vector3(x=0.0, y=0.0, z=0.0))

publishing #7: geometry_msgs.msg.Twist(linear=geometry_msgs.msg.Vector3(x=1.0, y=0.0, z=0.0), angular=geometry_msgs.msg.Vector3(x=0.0, y=0.0, z=0.0))

publishing #8: geometry_msgs.msg.Twist(linear=geometry_msgs.msg.Vector3(x=1.0, y=0.0, z=0.0), angular=geometry_msgs.msg.Vector3(x=0.0, y=0.0, z=0.0))

publishing #9: geometry_msgs.msg.Twist(linear=geometry_msgs.msg.Vector3(x=1.0, y=0.0, z=0.0), angular=geometry_msgs.msg.Vector3(x=0.0, y=0.0, z=0.0))

publishing #10: geometry_msgs.msg.Twist(linear=geometry_msgs.msg.Vector3(x=1.0, y=0.0, z=0.0), angular=geometry_msgs.msg.Vector3(x=0.0, y=0.0, z=0.0))
`

This continues until I halt the command. 

Any suggestions? Thanks!

**Comment by alsora** [comment, marked as answer]
@gjf2a your robot has a namespace, so you should publish to `/archangel/cmd_vel`

The full command would be:
```
ros2 topic pub /archangel/cmd_vel geometry_msgs/Twist '{linear: {x: 1.0, y: 0.0, z: 0.0}, angular: {x: 0.0, y: 0.0, z: 0.0}}'

```

**Comment by gjf2a** [comment]
Never mind - I had left off the `/archangel` part!
